dask/dataframe/io/parquet/fastparquet.py

```diff
@@ -19,9 +20,23 @@
 except ImportError:
     pass
 
-from .utils import _parse_pandas_metadata, _normalize_index_columns, _analyze_paths
+from .utils import (
+    _parse_pandas_metadata,
+    _normalize_index_columns,
+    _analyze_paths,
+    _flatten_filters,
+    _row_groups_to_parts,
+)
 from ..utils import _meta_from_dtypes
 from ...utils import UNKNOWN_CATEGORIES
+from ...methods import concat
+
+try:
+    # Required for distrubuted
+    import pickle
+except ImportError:
+    pickle = None
```

Comment:
Using pickle, when available, actually seems to improve performance for cases where the row-group objects need to be passed in the graph to handle hive-partitioning.